pkg/util/certificates/certificates.go

```diff
@@ -1,13 +1,133 @@
 package certificates
 
 import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
 	"crypto/x509"
+	"crypto/x509/pkix"
 	"encoding/pem"
+	"math/big"
+	"net"
 	"time"
 
 	"github.com/Dynatrace/dynatrace-operator/pkg/logd"
 )
 
+const (
+	intSerialNumberLimit = 128
+
+	certificatePemHeader = "CERTIFICATE"
+	privateKeyPemHeader  = "PRIVATE KEY"
+)
+
+var serialNumberLimit = new(big.Int).Lsh(big.NewInt(1), intSerialNumberLimit)
+
+type Certificate struct {
+	cert       *x509.Certificate
+	pk         *ecdsa.PrivateKey
+	signedCert []byte
+	signedPk   []byte
+}
+
+func New(domain string, altNames []string, ip string) (*Certificate, error) {
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, err
+	}
+
+	pk, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	netIp := net.ParseIP(ip)
+
+	cert := &x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			Country:            []string{"AT"},
```

Comment:
can we extract to constants all these strings?

Reply:
[Fixed](https://github.com/Dynatrace/dynatrace-operator/pull/3778/commits/8f912b88edf5f695f95af496983cced770f5ef14)